maas/resource_maas_zone_test.go

```diff
@@ -0,0 +1,118 @@
+package maas_test
+
+import (
+	"fmt"
+	"strings"
+	"terraform-provider-maas/maas/testutils"
+	"testing"
+
+	"github.com/canonical/gomaasclient/client"
+	"github.com/canonical/gomaasclient/entity"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestAccResourceMaasZone_basic(t *testing.T) {
+
+	var zone entity.Zone
+	name := acctest.RandomWithPrefix("tf-zone-")
+	description := "Test description"
+
+	checks := []resource.TestCheckFunc{
+		testAccMaasZoneCheckExists("maas_zone.test", &zone),
+		resource.TestCheckResourceAttr("maas_zone.test", "name", name),
+		resource.TestCheckResourceAttr("maas_zone.test", "description", description),
+	}
+
+	resource.ParallelTest(t, resource.TestCase{
+		Providers:    testutils.TestAccProviders,
```

Comment:
A precheck is missing here:

```go
PreCheck:     func() { testutils.PreCheck(t, nil) },
```

This function will check that `MAAS_API_URL` and `MAAS_API_KEY` are set before running the acceptance test.